app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -47,6 +49,13 @@ public CommentsViewModel(final @NonNull Environment environment) {
     this.client = environment.apiClient();
     this.currentUser = environment.currentUser();
 
+    final Observable<User> currentUser = Observable.merge(
+      this.currentUser.observable(),
+      this.currentUser.observable()
+        .compose(takeWhen(this.loginSuccess))
+        .flatMap(__ -> this.client.fetchCurrentUser().compose(neverError())).share()
```

Comment:
ah right right, since fetching the user is only contingent on the `loginSuccess`, not `currentUser.observable()` !